ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -110,6 +110,23 @@ def _process_visual_observation(
     return np.array(batched_visual, dtype=np.float32)
 
 
+def _raise_on_nan_and_inf(data: np.array, source: str) -> np.array:
+    # Check for NaNs or infs in the observations
+    # If there's a NaN in the observations, the np.mean() result will be NaN
+    # If there's an Inf (either sign) then the result will be Inf
+    # See https://stackoverflow.com/questions/6736590/fast-check-for-nan-in-numpy for background
+    # Note that a very large values (larger than sqrt(float_max)) will result in an Inf value here
+    # This is OK though, worst case it results in an unnecessary (but harmless) nan_to_num call.
+    d = np.mean(data)
+    has_nan = np.isnan(d)
+    has_inf = not np.isfinite(d)
+
+    if has_nan:
```

Comment:
Should we raise this error or just zero it out and print a warning? Some envs (e.g. Crawler) will produce occasional NaNs but the training will recover eventually.

Reply:
I don't think we should cover it up, since the NaNs will cause problems at inference too.